### PR TITLE
Upgrade Scala/Play/SBT

### PR DESCRIPTION
**Pull request by davidfurey**
## What does this change?

Upgrades to Java 11, Scala 2.13, SBT 1.9.6 and Play 2.9.1

Also swaps out scalacache-caffeine for scaffeine. The API for scalacache-caffeine had changed quite a bit with the bump to a Scala 2.13 compatible version, and it only has milestone releases for v1.0. In contrast, Scaffeine seems better supported and is used in other projects

## How to test

Deploy to CODE, check app behaves as expected.

## How can we measure success?

No one notices, but Snyk vulnerabilities drop

### CODE REVIEW

**Comment by davidfurey** [comment]
Deployed to CODE, seems to work, but perhaps warrants more extensive testing

**Review by Georges-GNM — APPROVED** [review]
Branch is on code and seems to be working as intended. 

Looks like scalacache was being used for something related to tags which, from my understanding of them in workflow, seem to be behaving as normal, so looks good to me 👍

File: app/controllers/Admin.scala [hunk 1]
```diff
@@ -250,7 +250,7 @@ class Admin(
       tagIds <- tagIdsFuture
       selectedSectionOption <- selectedSectionOptionFt
     } yield {
-      val capiKeyJson: Json = parser.parse(s"""{"CAPI_API_KEY": ${config.capiKey}}""").right.get
+      val capiKeyJson: Json = parser.parse(s"""{"CAPI_API_KEY": ${config.capiKey}}""").toOption.get
```
Comment (bryophyta) [inline app/controllers/Admin.scala:253]: @davidfurey this line seems to be throwing an error in CODE (I don't have Workflow admin permissions for PROD so can't check it there). It looks like this was the last commit to touch this code (though seems likely that the bug preceded this commit) -- do you have any context on what the code here's doing? Happy to try and raise a PR if it'd be useful.

Comment (bryophyta) [reply app/controllers/Admin.scala:253]: To reproduce:
https://workflow.code.dev-gutools.co.uk/admin/sections-and-tags

Logs: https://logs.gutools.co.uk/s/editorial-tools/app/discover#/doc/664d234d-de11-5f34-8ac5-9ee2089ac473/logstash-ed-tools-workflow-2024.11.21?id=cZ1JT5MBT0J6efU8a76d

**Pull request merged**